sdk/storage/azure-storage-file-datalake/tests/test_directory.py

```diff
@@ -0,0 +1,349 @@
+# coding: utf-8
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import pytest
+import unittest
+from datetime import datetime, timedelta
+
+from azure.core.exceptions import HttpResponseError, ResourceExistsError, ResourceNotFoundError
+from azure.storage.filedatalake import ContentSettings, DirectorySasPermissions, DataLakeDirectoryClient
+from azure.storage.filedatalake import DataLakeServiceClient, generate_directory_sas
+from testcase import (
+    StorageTestCase,
+    record,
+    TestMode
+
+)
+
+# ------------------------------------------------------------------------------
+TEST_DIRECTORY_PREFIX = 'directory'
+# ------------------------------------------------------------------------------
+
+
+class DirectoryTest(StorageTestCase):
+    def setUp(self):
+        super(DirectoryTest, self).setUp()
+        url = self._get_account_url()
+        self.dsc = DataLakeServiceClient(url, credential=self.settings.STORAGE_DATA_LAKE_ACCOUNT_KEY)
+        self.config = self.dsc._config
+
+        self.file_system_name = self.get_resource_name('filesystem')
+
+        if not self.is_playback():
+            file_system = self.dsc.get_file_system_client(self.file_system_name)
+            try:
+                file_system.create_file_system(timeout=5)
+            except ResourceExistsError:
+                pass
+
+    def tearDown(self):
+        if not self.is_playback():
+            try:
+                self.dsc.delete_file_system(self.file_system_name)
+                for file_system in self.dsc.list_file_systems():
+                    self.dsc.delete_file_system(file_system.name)
+            except:
+                pass
+
+        return super(DirectoryTest, self).tearDown()
+
+    # --Helpers-----------------------------------------------------------------
+    def _get_directory_reference(self, prefix=TEST_DIRECTORY_PREFIX):
+        directory_name = self.get_resource_name(prefix)
+        return directory_name
+
+    def _create_directory_and_get_directory_client(self, directory_name=None):
+        directory_name = directory_name if directory_name else self._get_directory_reference()
+        directory_client = self.dsc.get_directory_client(self.file_system_name, directory_name)
+        directory_client.create_directory()
+        return directory_client
+
+    def _create_file_system(self):
+        return self.dsc.create_file_system(self._get_file_system_reference())
+
+
+    # --Helpers-----------------------------------------------------------------
+
+    @record
+    def test_create_directory(self):
```

Comment:
Please log an issue to eventually migrate these tests to the new framework

Reply:
Sure! Created here ^_^ https://msazure.visualstudio.com/One/_workitems/edit/5703256